util/zstdframe/options.go

```diff
@@ -115,7 +146,12 @@ func getEncoder(opts ...Option) encoder {
 			zstd.WithZeroFrames(true),
 			zstd.WithEncoderLevel(eopts.level),
 			zstd.WithEncoderCRC(eopts.checksum),
-			zstd.WithLowerEncoderMem(eopts.lowMemory)))
+			zstd.WithLowerEncoderMem(eopts.lowMemory),
+		}
+		if eopts.maxWindowLog2 > 0 {
+			zopts = append(zopts, zstd.WithWindowSize(1<<eopts.maxWindowLog2))
+		}
+		enc = must.Get(zstd.NewWriter(nil, zopts...))
```

Comment:
likewise, can the compiler make this allocation free? the append would seem like it'd throw off escape analysis of a statically-sized zopts slice from above.

Reply:
This was fiddly, but I got it to play nice with escape analysis.